test/extensions/filters/http/common/compressor/compressor_filter_speed_test.cc

```diff
@@ -0,0 +1,237 @@
+#include "envoy/extensions/filters/http/compressor/v3/compressor.pb.h"
+
+#include "common/compressor/zlib_compressor_impl.h"
+
+#include "extensions/filters/http/common/compressor/compressor.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/stats/mocks.h"
+
+#include "benchmark/benchmark.h"
+#include "gmock/gmock.h"
+
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Compressors {
+
+class MockCompressorFilterConfig : public CompressorFilterConfig {
+public:
+  MockCompressorFilterConfig(
+      const envoy::extensions::filters::http::compressor::v3::Compressor& compressor,
+      const std::string& stats_prefix, Stats::Scope& scope, Runtime::Loader& runtime,
+      const std::string& compressor_name,
+      Envoy::Compressor::ZlibCompressorImpl::CompressionLevel level,
+      Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy strategy, int64_t window_bits,
+      uint64_t memory_level)
+      : CompressorFilterConfig(compressor, stats_prefix + compressor_name + ".", scope, runtime,
+                               compressor_name),
+        level_(level), strategy_(strategy), window_bits_(window_bits), memory_level_(memory_level) {
+  }
+
+  std::unique_ptr<Compressor::Compressor> makeCompressor() override {
+    auto compressor = std::make_unique<Compressor::ZlibCompressorImpl>();
+    compressor->init(level_, strategy_, window_bits_, memory_level_);
+    return compressor;
+  }
+
+  const Envoy::Compressor::ZlibCompressorImpl::CompressionLevel level_;
+  const Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy strategy_;
+  const int64_t window_bits_;
+  const uint64_t memory_level_;
+};
+
+using CompressionParams =
+    std::tuple<Envoy::Compressor::ZlibCompressorImpl::CompressionLevel,
+               Envoy::Compressor::ZlibCompressorImpl::CompressionStrategy, int64_t, uint64_t>;
+
+static std::vector<Buffer::OwnedImpl> generateChunks(const uint64_t chunk_count,
+                                                     const uint64_t chunk_size) {
+  std::vector<Buffer::OwnedImpl> vec;
+  vec.reserve(chunk_count);
+
+  Buffer::OwnedImpl data;
+  TestUtility::feedBufferWithRandomCharacters(data, chunk_size);
+
+  for (uint64_t i = 0; i < chunk_count; ++i) {
+    Buffer::OwnedImpl chunk;
+    chunk.add(data);
+    vec.push_back(std::move(chunk));
+  }
+
+  return vec;
+}
+
+static void compressWith(std::vector<Buffer::OwnedImpl> chunks, CompressionParams params,
```

Comment:
Shouldn't `chunks` be of type `const std::vetcor<Buffer::OwnedImpl>&` to make passing the vector as rvalue meaningful?

Reply:
ah yes (if you meant `const std::vector<Buffer::OwnedImpl>&&`).